### DISCUSSION TITLE
Expected performance on a Raspberry Pi 3

### DISCUSSION THREAD

**Discussion by superlou** [hiulit/Unofficial-Godot-Engine-Raspberry-Pi#30] [Q&A]
I'm using the Godot 3.5.2 build for a Raspberry Pi 3. I exported a very basic "game" from a Linux PC: text on a blank background (debug off). It runs on the Raspberry Pi but it takes about 3 minutes to load and start. The Godot 3.5.2 editor starts on the Raspberry Pi, but it takes around 10 minutes and is very laggy. Is that the expected performance on a Pi 3 or is it possible I have a configuration wrong?

**Comment by superlou** [comment]
Doing a bit better: wiped the existing OS with the latest one. Now I can run a very basic 2D project, but it's still a pretty low frame rate. The resolution is 1280x720 with scaling turned on.

**Comment by hiulit** [comment]
Hey, sorry for the late reply!
I've actually never used the editor on a Pi 3, so I guess it doesn't perfome as well as the Pi 4 :(
As for the export templates, I'm using [FRT](https://github.com/efornara/frt) and my game [Murtop](https://games.hiulit.com/murtop/) runs at 60 FPS on a Pi 3. I guess those are more optimized than mine.
Also, I did some tests a while back and you have to set the `stretch mode` to `viewport` to have decent FPS. If you use `disabled` or `2D`, its drops drastically (although `2D` it's acceptable, at ~ 30 FPS).

**Reply by TheTechRobo** [reply]
Pi 3 works fine for a GUI app for me (not much logic and it's running in low cpu mode). It's slow to start up, but nowhere near 3 minutes. More like 10 seconds for me. Maybe low CPU mode has something to do with that? :thinking:

**Reply by superlou** [reply]
> Pi 3 works fine for a GUI app for me (not much logic and it's running in low cpu mode). It's slow to start up, but nowhere near 3 minutes. More like 10 seconds for me. Maybe low CPU mode has something to do with that? 🤔

The 3 minute start time seems to have been largely corrected by switching by updating to a more recent version of Raspberry Pi OS. I'm going to try to set the stretch mode tonight to see if that gives me "lower resolution performance" at full-screen.

**Reply by TheTechRobo** [reply]
What's your startup time roughly now?

**Reply by superlou** [reply]
I just timed it at 15 seconds to get to the project manager and another 27 seconds to open the project for editing. It's pretty impressively quick. If I try to run the project, it creates a new window which then crashes before the project starts running. I've been exporting from my desktop using the export template and then running it on the RPi.
@hiulit, unfortunately performance is pretty similar when I try to scale up a window with stretch mode = viewport (though I definitely can see the stretching happening), about 10 fps. It's amazing it works so well, though! Hopefully one day Raspberry Pi 4s will be stocked again.

**Comment by superlou** [comment]
One thing that seems to be making a big difference is disabling the xcompmgr compositor. With a 1280x720 window, I'm getting 37 FPS, and when I maximize the window, it only goes down to 28 FPS. I wonder if there is a way for a Godot project to bypass the compositor.
